inc/user_templates/parameters/commonitilobjectparameters.class.php

```diff
@@ -0,0 +1,163 @@
+         new AttributeParameter("urgency", __("Ticket's urgency")),
+         new AttributeParameter("impact", __("Ticket's impact")),
+         new AttributeParameter("priority", __("Ticket's priority")),
+         new ObjectParameter("entity", new EntityParameters()),
+         new ObjectParameter("itilcategory", new ITILCategoryParameters()),
+         new ArrayParameter("requesters.users", 'user', new UserParameters(), __("Ticket's requesters (users)")),
+         new ArrayParameter("observers.users", 'user', new UserParameters(), __("Ticket's observers (users)")),
+         new ArrayParameter("technicians.users", 'user', new UserParameters(), __("Ticket's technicians (users)")),
+         new ArrayParameter("requesters.groups", 'group', new GroupParameters(), __("Ticket's requesters (groups)")),
+         new ArrayParameter("observers.groups", 'group', new GroupParameters(), __("Ticket's observers (groups)")),
+         new ArrayParameter("technicians.groups", 'group', new GroupParameters(), __("Ticket's technicians (groups)")),
+      ];
+   }
+
+   public function defineValues(CommonDBTM $commonitil): array {
+      /** @var CommonITILObject $commonitil  */
+
+      // Base values from ticket property
+      $values = [
+         'id'        => $commonitil->fields['id'],
+         'ref'       => "#" . $commonitil->fields['id'],
+         'link'      => $commonitil->getLink(),
+         'name'      => $commonitil->fields['name'],
+         'content'   => $commonitil->fields['content'],
+         'date'      => $commonitil->fields['date'],
+         'solvedate' => $commonitil->fields['solvedate'],
+         'closedate' => $commonitil->fields['closedate'],
+         'status'    => $commonitil::getStatus($commonitil->fields['status']),
+         'urgency'   => $commonitil::getUrgencyName($commonitil->fields['urgency']),
+         'impact'    => $commonitil::getImpactName($commonitil->fields['impact']),
+         'priority'  => $commonitil::getPriorityName($commonitil->fields['priority']),
+      ];
+
+      // Add ticket's entity
+      if ($entity = Entity::getById($commonitil->fields['entities_id'])) {
+         $entity_parameters = new EntityParameters();
+         $values['entity'] = $entity_parameters->getValues($entity);
+      }
+
+      // Add ticket's category
+      if ($itilcategory = ITILCategory::getById($commonitil->fields['itilcategories_id'])) {
+         $itilcategory_parameters = new ITILCategoryParameters();
+         $values['itilcategory'] = $itilcategory_parameters->getValues($itilcategory);
+      }
+
+      // Add requesters / observers / assigned data
+      $commonitil->loadActors();
+
+      $values['requesters'] = [
+         'users'  => [],
+         'groups' => [],
+      ];
+      $values['observers'] = [
+         'users'  => [],
+         'groups' => [],
+      ];
+      $values['technicians'] = [
+         'users'  => [],
+         'groups' => [],
```

Comment:
Suppliers and Contacts can also be assigned.

Reply:
I'll look into adding them

Reply:
Done in aeb21960b7162d18b45d391562b6ca788bdaa5a4.